### ISSUE TITLE
Routing information update as you move

### ISSUE THREAD

**Issue by nolispe280** [mapzen/ios#376]
### Description

I use the routing function in the SDK to get turn-by-turn information. By printing leg.maneuvers, I get all the maneuvers of the route ("turn right in 100m", "turn left in 50m") at once but no information on the coordinates from which I should consider them.
Is there a way of comparing your GPS location with the intersection coordinate so that you know you should use the right maneuver instruction?
Any help on this issue is welcome!

**Comment by msmollin** [comment]
Unfortunately that functionality is not currently available in the iOS SDK. However we are actively working on it and it should be ready later this quarter. It is possible to do manually however, and not super involved. Basically you would need to setup Core Location and receive location updates, and pass them into a routing engine that would then need a number of milestone checks.

We do have this functionality in Android currently, so you can base your system's logic off of that if you decide you want to go at this yourself. Take a look here https://github.com/mapzen/on-the-road_android/blob/master/library/src/main/java/com/mapzen/helpers/RouteEngine.java#L50
as that is the main entry point for when GPS changes occur.

**Comment by nolispe280** [comment]
Thank you Matt. I already have the location updates but the routing engine in the Mapzen-ios-sdk does not seem to have any function similar to the ones for Android (I don't fully understand what you mean by milestone checks). FYI, it seems to be possible to access "intersection coordinates" in the iOS sdk by using `route.legs[0].coordinates` but there is way too many points.

**Comment by msmollin** [comment]
That's correctly. Per my above response: _Unfortunately that functionality is not currently available in the iOS SDK. However we are actively working on it and it should be ready later this quarter. It is possible to do manually however, and not super involved._

**Comment by msmollin** [comment]
Milestone checks is one of the things you would need to implement. The Android implementation has examples of that (2 miles away, 1 mile away, etc.)

**Comment by msmollin** [comment]
The `coordinates` array is the raw lat/long coordinate pairs we use to draw the shape of the route on the map. The intersections are in there merely because they are part of the route polyline. Is there something in particular you are looking for intersection-wise?

**Comment by nolispe280** [comment]
Right, I already have implemented the milestone checks in my code but what is missing is the function that get the distance to the next instruction. To be more precise, only the coordinate of the intersection that corresponds to the next instruction is needed (I also already implemented a function that compute the distance to the next intersection).
Intersection-wise, I am interested in the intersection that corresponds to the instructions. (`leg.maneuvers` does not, unfortunately, have coordinates attached to the instructions)

**Comment by nolispe280** [comment]
Let me know if you need more explanation on the issue ;)

**Comment by msmollin** [comment]
Not to sound repetitive, but you should take a look at the Android code if you want that kind of functionality now. We're actively working on adding it to iOS but it won't be until later this quarter.

**Comment by nolispe280** [comment]
I understood what you wrote and I did look at the Android code. Unfortunately, the function that I need is the function `getDistanceToNextInstruction()` which is not detailed in the java script. I would love to 

> do manually

as you said but it seems like it is impossible at the moment without the coordinates of the intersections coresponding to the instructions.

**Comment by msmollin** [comment]
Finally getting back around to this (sorry about that). I'm not sure I'm fully understanding the issue. You receive back from Valhalla a list of maneuvers and voice instructions for those maneuvers. Each maneuver will correspond (roughly) to an intersection of a road. Is this not what you're looking for?

**Comment by nolispe280** [comment]
I don't understand what is so difficult about my inquiry (which I kind of fixed but it is not perfect). 
Simply put: there is no coordinates from which to start each maneuver on the list.

**Comment by nolispe280** [comment]
For 3 maneuvers (turn left, turn right, turn left), I would expect 3 coordinates.

**Comment by msmollin** [comment]
Ah, ok this is slightly different than the original request. That's why I was clarifying.

You are correct in that we don't expose raw GPS lat/longs directly in the `legs.maneuvers` object. This is likely an oversight on our part although honestly it hasn't been asked for until now. I will make sure it gets added as part of the ongoing work on it.

Thanks for taking the time to explain it.

**Comment by msmollin** [comment]
In the mean time, if you wanted to attempt to do it yourself, you could manually connect to Mapzen's turn by turn servers, retrieve the response, and then parse it using the available objects in our On The Road library. Once you get to the point you have the leg manuevers, you can use the `beginShapeIndex` (https://github.com/mapzen/on-the-road_ios/blob/master/OTRRouting/OTRRoutingResultManeuver.h#L115) from the manuever that is available to query the shape that got returned from the service for the GPS coordinate of the start of the manuever.

**Comment by nolispe280** [comment]
That has always been my one and only request ;) Thank you for taking the time to reply anyway. I already do have a way of doing it now manually but looking forward to seeing the option added to the ongoing work!
